src/main/java/com/example/Library/exception/BookExceptionHandler.java

```diff
@@ -0,0 +1,21 @@
+package com.example.Library.exception;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+public class BookExceptionHandler {
+    @ExceptionHandler({BookNotFoundException.class})
```

Comment:
To jest metoda obslugujaca wyjatek BookNotFoundException to po co jako parametr metody przyjmujesz Exception?
    public ResponseEntity<ErrorMessage> handleBookNotFound(BookNotFoundException, WebRequest request) {

Reply:
Done